### PR TITLE
FIX lost superadmin grade after edit user card

### PR DESCRIPTION
**Pull request by hregis**

### CODE REVIEW

File: htdocs/user/card.php [hunk 1]
```diff
@@ -450,11 +450,15 @@
 				// Do we update also ->entity ?
 				if (!empty($conf->multicompany->enabled)) {	// If multicompany is not enabled, we never update the entity of a user.
 					if (!empty($conf->global->MULTICOMPANY_TRANSVERSE_MODE)) {
-						$object->entity = 1; // all users are in master entity
+						if (GETPOST('superadmin', 'int')) {
+							$object->entity = 0;
```
Comment (eldy) [inline htdocs/user/card.php:454]: Relying on a property provided as a parameter to decide if an object must be into an admin entity or not means anybody, even not superadmin can set the superadmin flag to any user.

Comment (hregis) [reply htdocs/user/card.php:454]: @eldy 
I check this in the update function of the class user

**Comment by hregis** [comment]
@eldy
This correction is very important because a superadmin that edits its user profile or any superadmin profile downgrades the profile to a simple administrator!

File: htdocs/user/card.php [hunk 2]
```diff
@@ -448,19 +448,15 @@
 				$object->lang = GETPOST('default_lang', 'aZ09');
 
 				// Do we update also ->entity ?
-				if (!empty($conf->multicompany->enabled)) {	// If multicompany is not enabled, we never update the entity of a user.
-					if (!empty($conf->global->MULTICOMPANY_TRANSVERSE_MODE)) {
-						$object->entity = 1; // all users are in master entity
+				if (!empty($conf->multicompany->enabled && $user->entity == 0 && !empty($user->admin))) {	// If multicompany is not enabled, we never update the entity of a user.
+					if (GETPOST('superadmin', 'int')) {
```
Comment (eldy) [inline htdocs/user/card.php:452]: We should not rely on a value from GETPOST to force the entity.
According to your comment "// If multicompany is not enabled, we never update the entity of a user.", if this is true, why no just commenting the line 
$object->entity = 0;
If we must never update the field, we must just not set the value when we are in such a case.

**Pull request merged**